### DISCUSSION TITLE
Deserialize binary metadata to rust struct?

### DISCUSSION THREAD

**Discussion by mattoni** [hyperium/tonic#938] [Q&A]
Hi there,

I'm attempting to decode an error type on the client that is embedded in the metadata response that comes from an external gRPC service.

`res.metadata().get_bin("conflicterrorresponse-bin").unwrap()`

Which obviously gives a lot of binary data. Is it possible to deserialize this into a custom rust struct? 

Thanks!

**Comment by LucioFranco** [comment]
It returns bytes so you can plug that into any deser format you like.

**Comment by cxw620** [comment]
> Hi there,
> 
> I'm attempting to decode an error type on the client that is embedded in the metadata response that comes from an external gRPC service.
> 
> `res.metadata().get_bin("conflicterrorresponse-bin").unwrap()`
> 
> Which obviously gives a lot of binary data. Is it possible to deserialize this into a custom rust struct?
> 
> Thanks!

Do you have any solution for this problem? I'm new in rust programming and now confused by this problem, too.

**Reply by davidpdrsn** [reply]
bincode is a common one. serde_json as well

**Reply by mattoni** [reply]
This is essentially what I did. Ran the result thru bincode and deserialized into a struct.
